src/AbstractContainer.php

```diff
@@ -596,11 +586,19 @@ public function setExpirationHops($hops, $vars = null)
     }
 
     /** @inheritDoc */
-    public function getArrayCopy()
+    public function getArrayCopy(): array
     {
         $storage   = $this->verifyNamespace();
         $container = $storage[$this->getName()];
 
-        return $container instanceof ArrayObject ? $container->getArrayCopy() : $container;
+        if ($container instanceof ArrayObject) {
+            /** @var array<TKey, TValue> $array */
+            $array = $container->getArrayCopy();
+            return $array;
+        }
+
+        /** @var array<TKey, TValue> $array */
+        $array = $container;
+        return $array;
```

Comment:
This would be more readable with a temporary var before forcing the type:

```suggestion
        /** @var array<TKey, TValue> $array */
        $array = $container instanceof ArrayObject ? $container->getArrayCopy() : $container;
        
        return $array;
```

Loads of issues would be solved here if there were better types attributed to whatever `$this->getStorage()` returns. By the look of it, it's typically `StorageInterface<string, array<string, mixed>|ArrayObject<string, mixed>>`. (??)